spec/lite_spec_helper.rb

```diff
@@ -106,6 +108,12 @@
   end
 
   if SpecConfig.instance.ci?
+    # To avoid a buildup of connections, we periodically close and reconnect each client if the
```

Comment:
Previously, this was defined in a separate `around` hook in `spec_helper.rb` and without reconnecting the client. This worked because `close_stale_sockets!` previously eagerly reconnected sockets (as discussed offline). When `close_stale_sockets!` was updated to stop doing this, any time `close_local_clients` was called in the hook, tests would fail due to the pool being closed. I've changed `close_local_clients` to take an optional argument to reconnect the clients rather than just closing them. Unfortunately, this can take a while in some cases, and it sometimes pushes tests over the timeout boundary defined here. In order not to count the time it takes to reconnect the clients towards the timeout, I needed to move the implementation into the same `around` hook. To avoid reconnecting clients when only `lite_spec_helper` is included, I guarded it with the definition of a boolean constant in `spec_helper` defined right above the inclusion of `lite_spec_helper`. The client reconnections are done before the running of the test instead of after because [this assertion](https://github.com/mongodb/mongo-ruby-driver/blob/ad144e863879355372b324c3d48bf756372e5cd8/spec/mongo/client_construction_spec.rb#L69) failed due to the monitors starting when clients were reconnected after it.